### DISCUSSION TITLE
'>=' not supported between instances of 'int' and 'datetime.datetime'

### DISCUSSION THREAD

**Discussion by Elizmcc** [sqlalchemy/sqlalchemy#10048] [Usage Questions]
I'm couldn't resolve that error in my code. This is my code

import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.neighbors import NearestNeighbors
import statsmodels.api as sm
from datetime import datetime
import os

def get_matching_pairs(treated_df, non_treated_df, scaler=True):
    
    treated_x = treated_df.values
    non_tretaed_x = non_treated_df.values
    if scaler == True:
       scaler ==StandardScaler()
    if scaler:
        scaler.fit(treated_x)
        treated_x = scaler.transform(treated_x)
        non_tretaed_x = scaler.transform(non_tretaed_x)
    
    nbrs = NearestNeighbors(n_neighbors=1,algorithm='ball_tree').fit(non_tretaed_x)
    distances, indices = nbrs.kneighbors(treated_x)
    indices = indices.reshape(indices.shape[0])
    return indices

os.chdir('/Users/elizabethmcc/Desktop/MIT')
data=pd.read_excel('Cópia de 1st Alternative - Data (002).xlsx',sheet_name='Sheet1')

#code treated
data['Treated'] = 0
data.loc[data['Credit Access']=='Yes'] = 1

#code after column
data['After'] = 0
data.loc[data['After'] >= datetime(2023,3,1),'After'] = 1

#code interaction colum
data['Interaction'] = data['Treated']*data['After']

#Run OLS regression (not)
X2 = sm.add_constant(data[['Treated','After','Interaction']])
est = sm.OLS(data['Product Sold'],X2)
est2 = est.fit()
print(est2.summary())

And here is the data: [https://docs.google.com/spreadsheets/d/1YkSL2xlgUI5SrDg_dLDheWeQyA-_cTo2/edit?usp=drive_link&ouid=103332242051235504358&rtpof=true&sd=true](url)

**Comment by CaselIT** [comment]
Hi

I'm not sure where you wanted to post this, but this seems the wrong project
